From d102d2e00d6f232a5f767371bd70ecb744b8fd76 Mon Sep 17 00:00:00 2001
Message-ID: <d102d2e00d6f232a5f767371bd70ecb744b8fd76.1732837202.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1732837202.git.sharlatanus@gmail.com>
References: <cover.1732837202.git.sharlatanus@gmail.com>
From: "Artyom V. Poptsov" <poptsov.artyom@gmail.com>
Date: Sun, 24 Nov 2024 13:27:28 +0300
Subject: [PATCH 07/15] gnu: Add go-github-com-charmbracelet-x-exp-golden.

* gnu/packages/golang-xyz.scm (go-github-com-charmbracelet-x-exp-golden): New variable.

Change-Id: I7fa815433d9a83e4562bdc769312dd05e0a298da
Reviewed-by: Sharlatan Hellseher <sharlatanus@gmail.com>
---
 gnu/packages/golang-xyz.scm | 42 +++++++++++++++++++++++++++++++++++++
 1 file changed, 42 insertions(+)

diff --git a/gnu/packages/golang-xyz.scm b/gnu/packages/golang-xyz.scm
index d3b478dab0..81fd07f402 100644
--- a/gnu/packages/golang-xyz.scm
+++ b/gnu/packages/golang-xyz.scm
@@ -1589,6 +1589,48 @@ (define-public go-github-com-charmbracelet-x-ansi
 ECMA-48} specs.")
     (license license:expat)))
 
+(define-public go-github-com-charmbracelet-x-exp-golden
+  (package
+    (name "go-github-com-charmbracelet-x-exp-golden")
+    (version "0.0.0-20241121171228-5bc00623ea2f")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/charmbracelet/x")
+             (commit (go-version->git-ref version
+                                          #:subdir "exp/golden"))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "016s67690dr3w3an6m24q6f4vrmwpk0qd4akvvh1dzpfyf4khxd4"))))
+    (build-system go-build-system)
+    (arguments
+     (list
+      #:import-path "github.com/charmbracelet/x/exp/golden"
+      #:unpack-path "github.com/charmbracelet/x/"
+      #:phases
+      #~(modify-phases %standard-phases
+          (add-before 'check 'pre-check
+            (lambda* (#:key import-path #:allow-other-keys)
+              ;; Tests need to write to that files.
+              (with-directory-excursion (string-append "src/" import-path)
+                (make-file-writable "testdata/TestRequireEqualUpdate.golden")
+                (make-file-writable "testdata/TestRequireEqualNoUpdate.golden"))))
+          (add-after 'check 'post-check
+            (lambda* (#:key import-path #:allow-other-keys)
+              (with-directory-excursion (string-append "src/" import-path)
+                ;; Remove modified testdata just in case.
+                (delete-file-recursively "testdata")))))))
+    (propagated-inputs
+     (list go-github-com-aymanbagabas-go-udiff))
+    (home-page "https://github.com/charmbracelet/x")
+    (synopsis "Verify @code{.golden} file equality")
+    (description
+     "Golden files (@code{.golden}) contain the raw expected output of
+tests,which can contain control codes and escape sequences.  @code{golden}
+package provides an API for comparing Golden files.")
+    (license license:expat)))
+
 (define-public go-github-com-chzyer-logex
   (package
     (name "go-github-com-chzyer-logex")
-- 
2.46.0

